src/engine/source/kvdb/src/kvdbManager.cpp

```diff
@@ -0,0 +1,181 @@
+            {
+                if (!addDB(DB))
+                {
+                    delete DB;
+                    // TODO DEBUG Log
+                }
+            }
+            else
+            {
+                delete DB;
+                // TODO DEBUG Log
+            }
+        }
+    }
+
+    if (std::filesystem::exists(FOLDER + "legacy"))
+    {
+        for (const auto &cdbfile :
+             std::filesystem::directory_iterator(FOLDER + "legacy"))
+        { // Read it from the config file?
+            if (cdbfile.is_regular_file())
+            {
+                if (createDBfromCDB(cdbfile.path(), false))
+                {
+                    // TODO Remove files once synced
+                    // std::filesystem::remove(cdbfile.path())
+                }
+            }
+        }
+    }
+}
+
+KVDBManager::~KVDBManager()
+{
+    for (DBMap::iterator it = available_kvdbs.begin();
+         it != available_kvdbs.end();
+         it++)
+    {
+        delete it->second;
+    }
+    available_kvdbs.clear();
+}
+
+bool KVDBManager::createDB(const std::string &name, bool replace)
+{
+    ROCKSDB::Options createOptions;
+    createOptions.IncreaseParallelism();
+    createOptions.OptimizeLevelStyleCompaction();
+    createOptions.create_if_missing = true;
+    createOptions.error_if_exists = !replace;
+
+    ROCKSDB::DB *db;
+    ROCKSDB::Status s = ROCKSDB::DB::Open(createOptions, FOLDER + name, &db);
+    if (!s.ok())
+    {
+        // LOG(ERROR) << "[" << __func__ << "]" << " couldn't open db file,
+        // error: " << s.ToString() << std::endl;
+        return false;
+    }
+    s = db->Close(); // TODO: We can avoid this unnecessary close making a more
+                     // complex KVDB constructor that creates DBs
+                     // or if receives a CFHandler and a CFDescriptor vector.
+    if (!s.ok())
+    {
+        // LOG(WARNING) << "[" << __func__ << "]" << " couldn't close db file,
+        // error: " << s.ToString() << std::endl;
+        return false;
+    }
+
+    KVDB *kvdb = new KVDB(name, FOLDER);
+    if (!addDB(kvdb))
+    {
+        delete kvdb;
+        return false;
+    }
+
+    return true;
+}
+
+bool KVDBManager::createDBfromCDB(const std::filesystem::path &path,
+                                  bool replace)
+{
+    std::ifstream CDBfile(path);
+    if (!CDBfile.is_open())
+    {
+        // TODO Log Error
+        return false;
+    }
+
+    if (!createDB(path.stem(), replace))
+    {
+        // TODO Log Error
+        CDBfile.close();
+        return false;
+    }
+    KVDB &kvdb = getDB(path.stem());
+    if (kvdb.getState() != KVDB::State::Open)
+    {
+        // TODO Log Error
+        CDBfile.close();
+        return false;
+    }
+
+    std::string line;
+    while (getline(CDBfile, line))
+    {
+        line.erase(remove_if(line.begin(), line.end(), isspace), line.end());
+        auto KV = utils::string::split(line, ':');
+        kvdb.write(KV[0], KV[1]);
+    }
```

Comment:
with c++17 we can do
```suggestion
    while (std::string line; getline(CDBfile, line))
    {
        line.erase(remove_if(line.begin(), line.end(), isspace), line.end());
        auto KV = utils::string::split(line, ':');
        kvdb.write(KV[0], KV[1]);
    }
```
This way the string has the explicit 'lifetime' it needs

Reply:
can't make it work with the **while()** clause (but **if()** has no problem) 
```
 error: expected initializer before ‘;’ token
  116 |     while (std::string line; getline(CDBfile, line))
      |                            ^
```